test/integration/minikube_profile_test.go

```diff
@@ -0,0 +1,52 @@
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"os/exec"
+	"testing"
+
+	"k8s.io/minikube/pkg/minikube/config"
+)
+
+// profileJSON is the output of `minikube profile list -ojson`
+type profileJSON struct {
+	Valid   []config.Profile `json:"valid"`
+	Invalid []config.Profile `json:"invalid"`
+}
+
+func TestMinikubeProfile(t *testing.T) {
+	// 1. Setup two minikube cluster profiles
+	profiles := []string{UniqueProfileName("first"), UniqueProfileName("second")}
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(40))
+	for _, p := range profiles {
+		defer CleanupWithLogs(t, p, cancel)
+		_, err := Run(t, exec.CommandContext(ctx, Target(), "start", "-p", p))
+		if err != nil {
+			t.Errorf("test pre-condition failed: minikube start -p " + p + " failed: " + err.Error())
```

Comment:
```suggestion
			t.Fatalf("test pre-condition failed: minikube start -p " + p + " failed: " + err.Error())
```
If the start fails we should kill the test, as it's impossible for the test to pass at this point